### PR TITLE
Upload tool adapter

### PR DESCRIPTION
**Pull request by dqgorelick**
Depends on work completed for the new OpenAQ Upload Tool: https://github.com/openaq/openaq-upload/pull/24

### CODE REVIEW

File: adapters/upload_tool.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    parameter: a.parameter,
+    location: a.location,
+    country: a.country,
+    value: parseFloat(a.value),
+    unit: a.unit,
+    date: {
+      utc: a.date_utc,
+      local: a.date_local
+    },  
+    sourceName: a.sourceName,
```
Comment (dqgorelick) [inline adapters/upload_tool.js:29]: Change this to be sourceName: source.name (add this argument)

File: adapters/upload_tool.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  return new Promise((resolve, reject) => {
+    let results = []
+    s3.listObjects((params), async function (e, data) {
+        if(e) {
+          reject(e)
+        }
+        for (let i = 0; i < data.Contents.length; i++) {
+          try {
+            const fileParams = {
+              Bucket: 'upload-tool-bucket-dev',
```
Comment (dqgorelick) [inline adapters/upload_tool.js:102]: Note: this currently expects the final S3 bucket for the files to exist in `upload-tool-bucket-dev`. This bucket needs to be changed to match what is in the [Upload tool](https://github.com/openaq/openaq-upload/pull/24) repo for production stage.

File: adapters/upload_tool.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+'use strict';
+
+import { S3 } from 'aws-sdk';
+import csv from 'csv-parser'
+import log from '../lib/logger';
+
+// The S3 bucket containing the data is in a different region
+const s3 = new S3({region: 'us-west-2'});
+
+const UPLOAD_TOOL_BUCKET = 'upload-tool-bucket-dev'
```
Comment (dqgorelick) [inline adapters/upload_tool.js:15]: Note: this currently expects the final S3 bucket for the files to exist in upload-tool-bucket-dev. This bucket needs to be changed to match what is in the [Upload tool](https://github.com/openaq/openaq-upload/pull/24) repo for production stage.

**Review by sruti — CHANGES_REQUESTED** [review]
hi

File: adapters/upload_tool.js [hunk 4]
```diff
@@ -102,7 +102,7 @@ const readS3Files = function (params, source) {
             Key: data.Contents[i].Key
           };
           const result = await readFile(fileParams, source);
-          results.push(result);
+          results = results.concat(result);
```
Comment (dqgorelick) [inline adapters/upload_tool.js:105]: nice catch!

**Comment by MichalCz** [comment]
Hi,

I added this PR to make use of streaming and it actually does solve the issues that cause the errors.

**Comment by dqgorelick** [comment]
Hi @MichalCz thank you for taking a look. It's really interesting seeing how you use the generator function! 

Merging :shipit:

**Pull request closed without merging**